osu.Game/Beatmaps/Drawables/BeatmapGroup.cs

```diff
@@ -48,14 +51,20 @@ public BeatmapGroupState State
                             SelectedPanel.State = PanelSelectedState.Selected;
                         break;
                     case BeatmapGroupState.Collapsed:
+                    case BeatmapGroupState.Hidden:
+                        if (state == BeatmapGroupState.Hidden && state != value)

```

Comment:
`Header.Alpha = state == BeatmapGroupState.Hidden ? 0 : 1;`
put this outside the switch and remove all the conditionals you added inside it.